x/merkledb/view.go

```diff
@@ -271,8 +271,8 @@ func (v *view) calculateNodeIDsHelper(n *node) ids.ID {
 	// We use [wg] to wait until all descendants of [n] have been updated.
 	var wg sync.WaitGroup
 
-	for childIndex := range n.children {
-		childEntry := n.children[childIndex]
+	for childIndex, childEntry := range n.children {
+		childEntry := childEntry // New variable so goroutine doesn't capture loop variable.
```

Comment:
this should not be required anymore in 1.22 IIUC (see https://go.dev/blog/loopvar-preview) However this will still be fine